### PR TITLE
Make session_start_time required

### LINKED ISSUE #354: can we not automatically assign this?
https://github.com/catalystneuro/nwb-conversion-tools/blob/d1c78a9048f10d635379195acff645bddd21647e/nwb_conversion_tools/utils/spike_interface.py#L1270

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
fix #354 

## Motivation
For a long time, we've had `session_start_time` be an optional value throughout our pipelines, with a default value of `datetime(1970,1,1)`. However, it's becoming clear that as a best practice, this has potential to be completely missed by users which can result in an incorrect NWBFile uploaded to DANDI.

## How to test the behavior?
To-Do:
- [x] Need to add explicit tests and catches for the assertions for these changes.
- [x] Need to fix everything this breaks by default (i.e., tests for interfaces that don't auto-fill this)

## Checklist

- [X] Have you checked our [Contributing](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/docs/contribute.rst) document?
- [X] Have you ensured the PR description clearly describes the problem and solutions?
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/nwb-conversion-tools/pulls) for the same change?
- [X] If this PR fixes an issue, is the first line of the PR description `fix #XXX` where `XXX` is the issue number?

### CODE REVIEW

File: nwb_conversion_tools/nwbconverter.py [hunk 1]
```diff
@@ -113,6 +114,7 @@ def validate_conversion_options(self, conversion_options):
     def run_conversion(
         self,
         metadata: Optional[dict] = None,
+        session_start_time: Optional[datetime] = None,
```
Comment (bendichter) [inline nwb_conversion_tools/nwbconverter.py:117]: why add this as an additional arg?

Comment (CodyCBakerPhD) [reply nwb_conversion_tools/nwbconverter.py:117]: Fair point, suppose it should just be an extra arg for the standalone write functions in utils

**Comment by CodyCBakerPhD** [comment]
Finished debugging and adding tests; actual code changes are +11/-22, the extra code in the changelog is from all the places in the tests that now require `session_start_time` to be specified in place of relying on the default.

File: tests/test_internals/test_si.py [hunk 2]
```diff
@@ -124,21 +124,36 @@ def check_si_roundtrip(self, path: FilePathType):
     def test_write_recording(self):
         path = self.test_dir + "/test.nwb"
 
-        write_recording(self.RX, path)
+        write_recording(
+            self.RX,
+            path,
+            metadata=dict(NWBFile=dict(session_start_time=datetime.now().astimezone().strftime("%Y-%m-%dT%H:%M:%S"))),
```
Comment (bendichter) [inline tests/test_internals/test_si.py:130]: this dict can be declared earlier and used multiple times

File: tests/test_internals/test_tools.py [hunk 3]
```diff
@@ -26,3 +29,13 @@ def test_get_module(self):
         assert isinstance(mod_2, ProcessingModule)
         assert mod_2.description == description_2
         self.assertWarns(UserWarning, get_module, **dict(nwbfile=nwbfile, name=name_1, description=description_2))
+
+    def test_make_nwbfile_from_metadata(self):
+        with self.assertRaisesWith(
+            exc_type=AssertionError,
+            exc_msg=(
+                "'session_start_time' was not found auto-populated in metadata['NWBFile']! "
```
Comment (bendichter) [inline tests/test_internals/test_tools.py:37]:
```suggestion
                "'session_start_time' was not found in metadata['NWBFile']! "
```
it doesn't necessarily need to be auto-populated

File: tests/test_internals/test_tools.py [hunk 4]
```diff
@@ -26,3 +29,13 @@ def test_get_module(self):
...
+                "Please add the correct start time of the session in ISO8601 format to this key of the metadata."
```
Comment (bendichter) [inline tests/test_internals/test_tools.py:38]:
```suggestion
                "Please add the correct start time of the session in ISO8601 format (%Y-%m-%dT%H:%M:%S) to this key of the metadata."
```

**Review by bendichter — APPROVED** [review]

**Pull request merged**